internal/controllers/device_config_controller.go

```diff
@@ -53,113 +104,109 @@ type PowerConfig struct {
 	} `json:"wake_trigger"`
 }
 
-// NewDeviceConfigController constructor
-func NewDeviceConfigController(settings *config.Settings, logger *zerolog.Logger) DeviceConfigController {
-	return DeviceConfigController{
-		Settings: settings,
-		log:      logger,
-	}
+type DBCFile struct {
+	FilePath     string
+	TemplateName string
+	Version      string
+	CreatedAt    string
+	UpdatedAt    string
 }
 
-// / GetPIDConfig godoc
-// @Description  Retrieve the PID configuration based on a given VIN
+//Endpoints:
+
+// GetPIDSByTemplate godoc
+// @Description  Retrieves a list of PID configurations from the database given a template name
 // @Tags         vehicle-signal-decoding
 // @Produce      json
-// @Success      200 {object} PIDConfig
-// @Param        vin  path   string  true   "vehicle identification number (VIN)"
-// @Router       /device-config/:vin/pid [get]
-func (d *DeviceConfigController) GetPIDConfig(c *fiber.Ctx) error {
-	vin := c.Params("vin")
-	pidConfig := PIDConfig{
-		Name:            vin,
-		Header:          2015,
-		Mode:            9,
-		PID:             2,
-		Formula:         "ascii: 3|17 X",
-		IntervalSeconds: 5,
+// @Success      200 {array} PIDConfig "Successfully retrieved PID Configurations"
+// @Param        template_name  path   string  true   "template name"
+// @Router       /device-config/:template_name/pids [get]
+func (d *DeviceConfigController) GetPIDsByTemplate(c *fiber.Ctx) error {
+	templateName := c.Params("template_name")
+	var pids []PIDConfig
+
+	// Query the database to get the PIDs based on the template name
+	query := "SELECT * FROM pid_configs WHERE template_name = $1"
+	rows, err := d.db.Query(query, templateName)
+	if err != nil {
+		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
+			"error": "Failed to retrieve PID data",
+		})
+	}
+	defer rows.Close()
+
+	for rows.Next() {
```

Comment:
let's leverage sqlboiler instead of native sql stuff

Reply:
done for all handlers